analyzers/src/SonarAnalyzer.Common/Extensions/SyntaxTreeExtensions.cs

```diff
@@ -39,6 +41,22 @@ public static bool IsGenerated(this SyntaxTree tree, GeneratedCodeRecognizer gen
         // Hotpath: Don't use cache.GetOrAdd that takes a factory method. It allocates a delegate which causes GC preasure.
         return cache.TryGetValue(tree, out var isGenerated)
             ? isGenerated
-            : cache.GetOrAdd(tree, generatedCodeRecognizer.IsConsideredGenerated(tree));
+            : cache.GetOrAdd(tree, generatedCodeRecognizer.IsGenerated(tree));
+    }
+
+    public static bool IsConsideredGenerated(this SyntaxTree tree, GeneratedCodeRecognizer generatedCodeRecognizer, Compilation compilation) =>
+        IsRazorAnalysisEnabled()
+            ? IsGenerated(tree, generatedCodeRecognizer, compilation) && !GeneratedCodeRecognizer.IsRazorGeneratedFile(tree)
+            : IsGenerated(tree, generatedCodeRecognizer, compilation);
+
+    private static bool IsRazorAnalysisEnabled()
+    {
+        var value = Environment.GetEnvironmentVariable(EnableRazorAnalysisVariable);
+
+        if (value != null && bool.TryParse(value, out var result))
+        {
+            return result;
+        }
+        return false;
```

Comment:
Since we want to be on master without shipping the feature now we set the default value to false. Once we ship the full implementation of the feature the environment variable will be removed and replaced by a parameter that indeed has `true` as the default value. See:https://github.com/SonarSource/sonar-dotnet/issues/7882